Update holiday.py to reflect that MLK Jr. Day was first observed in 1986.

clearf opened the pull request:
closes #9760

Fixing MLK day not observed before 1986, per https://github.com/pydata/pandas/issues/10278#issuecomment-109002244

clearf commented:
Hi, 
I also noticed what appear to be two additional inconsistencies. 
* Memorial day is the last Monday in May. Code as before would not work if May 31st were a Monday. 
* New Years Day is not offset to the nearest workday in the US calendar as far as I can tell.

rockg commented:
Memorial day is already fixed in #9763 but it needs a test before it can be pushed.  Maybe it's best to incorporate the test there into this PR.  Also, all your other changes need tests.

rockg commented:
Also, New Year's Day is offset to the nearest workday for US Holidays.  https://www.opm.gov/policy-data-oversight/snow-dismissal-procedures/federal-holidays/.  Every other calendar used for holiday work uses a similar convention and so I think it makes sense to leave the rule as it is.  These rules are mostly meant to be examples and cover the standard use cases so it makes sense to leave New Years going to nearest_workday.

clearf commented:
OK, thanks. That makes a lot of sense. And you're right re: federal
holidays. I misread the spec. And they are a good example.

If Memorial Day is fixed elsewhere, then it seems like only MLK is the only
change that should actually be fixed. Sorry for the churn.

Thanks,
Chris


On Fri, Jun 5, 2015 at 8:58 AM, rockg <notifications@github.com> wrote:

> Also, New Year's Day is offset to the nearest workday for US Holidays.
> https://www.opm.gov/policy-data-oversight/snow-dismissal-procedures/federal-holidays/.
> Every other calendar used for holiday work uses a similar convention and so
> I think it makes sense to leave the rule as it is. These rules are mostly
> meant to be examples and cover the standard use cases so it makes sense to
> leave New Years going to nearest_workday.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10282#issuecomment-109284674>.
>



-- 
Chris Clearfield

jreback commented:
@clearf can you add some tests?

clearf commented:
@jreback sorry for the delay (camping). Happy to add some tests.

Some pointers to the test/build process would be great.

Thanks,
Chris


On Fri, Jun 26, 2015 at 4:33 PM, jreback <notifications@github.com> wrote:

> @clearf <https://github.com/clearf> can you add some tests?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10282#issuecomment-115917681>.
>



-- 
Chris Clearfield

jreback commented:
http://pandas.pydata.org/pandas-docs/stable/contributing.html

clearf commented:
Great, thanks.

On Fri, Jul 3, 2015 at 6:09 PM, jreback <notifications@github.com> wrote:

> http://pandas.pydata.org/pandas-docs/stable/contributing.html
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10282#issuecomment-118447458>.
>



-- 
Chris Clearfield

clearf commented:
@jreback Per above, added tests for MLK day. Also added tests for Memorial day fix, though my implementation is slightly different than that referenced in https://github.com/pydata/pandas/pull/9763

Sorry it took so long.

jreback commented on pandas/tests/test_tseries.py:
```diff
@@ -739,6 +739,32 @@ def test_get_period_field_raises_on_out_of_range(self):
+class TestFederalHolidayCalendar(tm.TestCase):
+
+    def test_no_mlk_before_1984(self):
+        from pandas.tseries.holiday import USMartinLutherKingJr, AbstractHolidayCalendar
```
you can put the imports at the top of the file with the rest of them.

jreback commented on pandas/tests/test_tseries.py:
```diff
@@ -739,6 +739,32 @@ def test_get_period_field_raises_on_out_of_range(self):
+        self.assertEqual(holidays, [datetime.datetime(1986, 1, 20, 0, 0), datetime.datetime(1987, 1, 19, 0, 0)])
+
+    def test_memorial_day(self):
+        from pandas.tseries.holiday import USMemorialDay, AbstractHolidayCalendar
```
add the issue number here as a comment

jreback commented:
looks good. couple of minor comments, then pls squash. ping when green.

clearf commented:
@jreback, OK, comments added and imports moved. All checks passed. 

And my apologies: I squashed my most recent commits, but could not figure out how to squash my commits from early June that predated the merge. Let me know if you know how to do this and I can fix.

jreback commented:
contributing docs here: http://pandas.pydata.org/pandas-docs/stable/contributing.html

pls rebase/squash

clearf commented:
Ok, thanks.

On Mon, Aug 10, 2015 at 04:04 Jeff Reback <notifications@github.com> wrote:

> contributing docs here:
> http://pandas.pydata.org/pandas-docs/stable/contributing.html
>
> pls rebase/squash
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10282#issuecomment-129408546>.
>

jreback commented on doc/source/whatsnew/v0.17.0.txt:
```diff
@@ -561,6 +561,7 @@ Bug Fixes
 - Bug in ``Table.select_column`` where name is not preserved (:issue:`10392`)
 - Bug in ``offsets.generate_range`` where ``start`` and ``end`` have finer precision than ``offset`` (:issue:`9907`)
 - Bug in ``pd.rolling_*`` where ``Series.name`` would be lost in the output (:issue:`10565`)
+- Bug in ``USFederalHolidayCalendar`` where ``USMemorialDay`` and ``USMartinLutherKingJr`` were incorrect (:issue:`10278`)
```
add on #9760 as well here

clearf commented:
@jreback Jeff, I rebased all my comments and pushed, but somehow have managed to get in a bunch of other commits from upstream. I'm not sure how to squash these. Sorry, this is my first time doing this.

jreback commented:
you need to ``git rebase -i origin/master``

clearf committed:
BUG: Fix MLK and Memorial day date representations

clearf commented:
@jreback And... I think we're there. Thanks for the guidance/patience.

jreback commented:
merged via https://github.com/pydata/pandas/commit/8ccdf4821f2dcac2b2c0f210d07679b5178f0211

thanks!

clearf commented:
Small victories. :)
On Tue, Aug 18, 2015 at 04:09 Jeff Reback <notifications@github.com> wrote:

> Closed #10282 <https://github.com/pydata/pandas/pull/10282>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pydata/pandas/pull/10282#event-385240783>.
>
